backend/src/main/java/meowhub/backend/jpa_buddy/UserRelation.java

```diff
@@ -0,0 +1,74 @@
+package meowhub.backend.jpa_buddy;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.ColumnDefault;
+import org.hibernate.annotations.OnDelete;
+import org.hibernate.annotations.OnDeleteAction;
+
+import java.time.LocalDate;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "USER_RELATIONS", schema = "mh_user_relations")
+public class UserRelation {
+    @Id
+    @Size(max = 36)
+    @ColumnDefault("sys_guid()")
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column(name = "ID", nullable = false, length = 36)
+    private String id;
+
+    @NotNull
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @OnDelete(action = OnDeleteAction.RESTRICT)
+    @JoinColumn(name = "RELATION_TYPE_ID", nullable = false)
+    private RelationType relationType;
+
+    @NotNull
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @OnDelete(action = OnDeleteAction.RESTRICT)
+    @JoinColumn(name = "SENDER_ID", nullable = false)
+    private User sender;
+
+    @NotNull
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @OnDelete(action = OnDeleteAction.RESTRICT)
+    @JoinColumn(name = "RECEIVER_ID", nullable = false)
+    private User receiver;
+
+    @NotNull
+    @Column(name = "SEND_DATE", nullable = false)
+    private LocalDate sendDate;
+
+    @NotNull
+    @Column(name = "ANSWER_DATE", nullable = false)
```

Comment:
Annotation @NotNull and nullable attribute is redundant. Maybe let's choose how we are going to set if something is not nullable